src/main/java/net/dv8tion/jda/api/requests/restaction/CommandCreateAction.java

```diff
@@ -51,234 +54,84 @@
-     *             <li>If more than 25 options are provided.</li>
-     *             <li>If null is provided</li>
-     *         </ul>
-     *
-     * @return The CommandCreateAction instance, for chaining
-     */
     @Nonnull
+    @Override
     @CheckReturnValue
     CommandCreateAction addOptions(@Nonnull OptionData... options);
 
-    /**
-     * Add up to 25 options to this command.
-     *
-     * <p>Required options must be added before non-required options!
-     *
-     * @param  options
-     *         The options to add
-     *
-     * @throws IllegalArgumentException
-     *         <ul>
-     *             <li>If you try to mix subcommands/options/groups in one command.</li>
-     *             <li>If the option type is {@link OptionType#SUB_COMMAND} or {@link OptionType#SUB_COMMAND_GROUP}.</li>
-     *             <li>If this option is required and you already added a non-required option.</li>
-     *             <li>If more than 25 options are provided.</li>
-     *             <li>If null is provided</li>
-     *         </ul>
-     *
-     * @return The CommandCreateAction instance, for chaining
-     */
     @Nonnull
+    @Override
     @CheckReturnValue
     default CommandCreateAction addOptions(@Nonnull Collection<? extends OptionData> options)
     {
-        Checks.noneNull(options, "Option");
-        return addOptions(options.toArray(new OptionData[0]));
+        return (CommandCreateAction) SlashCommandData.super.addOptions(options);
     }
 
-    /**
-     * Add one option to this command.
-     *
-     * <p>Required options must be added before non-required options!
-     *
-     * @param  type
-     *         The {@link OptionType}
-     * @param  name
-     *         The lowercase option name, 1-32 characters
-     * @param  description
-     *         The option description, 1-100 characters
-     * @param  required
-     *         Whether this option is required (See {@link OptionData#setRequired(boolean)})
-     *
-     * @throws IllegalArgumentException
-     *         <ul>
-     *             <li>If you try to mix subcommands/options/groups in one command.</li>
-     *             <li>If the option type is {@link OptionType#SUB_COMMAND} or {@link OptionType#SUB_COMMAND_GROUP}.</li>
-     *             <li>If this option is required and you already added a non-required option.</li>
-     *             <li>If more than 25 options are provided.</li>
-     *             <li>If null is provided</li>
-     *         </ul>
-     *
-     * @return The CommandCreateAction instance, for chaining
-     */
     @Nonnull
+    @Override
+    @CheckReturnValue
+    default CommandCreateAction addOption(@Nonnull OptionType type, @Nonnull String name, @Nonnull String description, boolean required, boolean autoComplete)
+    {
+        return (CommandCreateAction) SlashCommandData.super.addOption(type, name, description, required, autoComplete);
+    }
```

Comment:
You can't call these methods for context commands, because this rest-action is only returned for slash commands. To create a context command you use `guild.upsertCommand(Commands.user(name))` which returns `RestAction<Command>` instead.